### ISSUE TITLE
Receiving "Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)" when deploying our app

### ISSUE THREAD

**Issue by mcdurdin** [projectkudu/kudu#3042]
This issue has been moved over from https://developercommunity.visualstudio.com/content/problem/753520/receiving-error-error-failed-to-deploy-web-package.html

We are sporadically receiving `"Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)"` when deploying our app, with no other useful details that I can find. Following this, I can usually log in to dev.azure.com and start a new deployment which runs successfully. I just can't find any more detail about what the root issue is or how we can address this problem.

> For deployment issues, please provide us with the following information:
> ---------------------------------------------------------------

### Repro steps.
> your project built successfully on your dev machine but failed on Azure?
> please write down your build tools and their versions (ie `Msbuild 15.1.0.0`)

The project builds successfully both locally and on Azure (Node.js backend / Angular+Typescript front end). It fails only in the Release pipeline.

### Project structures.
> in order to reproduce your issue at our end we need a simple 
> [github repository](https://github.com/projectkudu/kudu/wiki/Using-a-git-repo-to-report-an-issue)
> that highlights structure of the project

See below; the site is straightforward in structure and documented, hopefully fairly minimal as it stands. The site won't run without specific keys for build.palaso.org (and perhaps github.com, although any user's key should work here); I can supply test keys if required.

### The log/error given by the failure.
> Normally this include a stack trace, error code and some more information.

```
2019-09-29T20:01:05.5384672Z ##[section]Starting: Deploy Azure App Service
2019-09-29T20:01:05.5523287Z ==============================================================================
2019-09-29T20:01:05.5523400Z Task         : Azure App Service deploy
2019-09-29T20:01:05.5523499Z Description  : Deploy to Azure App Service a web, mobile, or API app using Docker, Java, .NET, .NET Core, Node.js, PHP, Python, or Ruby
2019-09-29T20:01:05.5523568Z Version      : 3.4.31
2019-09-29T20:01:05.5523636Z Author       : Microsoft Corporation
2019-09-29T20:01:05.5523703Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-rm-web-app-deployment
2019-09-29T20:01:05.5523797Z ==============================================================================
2019-09-29T20:01:06.3287589Z Got connection details for Azure App Service:'status-keyman-com'
2019-09-29T20:01:07.4387353Z Package deployment using ZIP Deploy initiated.
2019-09-29T20:03:25.3932504Z ##[error]Failed to deploy web package to App Service.
2019-09-29T20:03:25.3976382Z ##[error]Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)
2019-09-29T20:03:29.6500803Z ##[warning]Error: Failed to update deployment history. Error: Conflict (CODE: 409)
2019-09-29T20:03:29.6646084Z ##[section]Finishing: Deploy Azure App Service
```

### Debug your Azure website remotely.
> it is recommanded that you share your Web App name, directly or [indirectly](https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly)
> we can take a look at what's going on.

The GH repo is https://github.com/keymanapp/status.keyman.com; live site is https://status.keyman.com

### Mention any other details that might be useful.

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by suwatch** [comment]
@sanchitmehta  please investigate this linux deployment issue.

@mcdurdin  the 409 conflict indicated ongoing deployment.   Please check if you have other deployments active during such time.

**Comment by mcdurdin** [comment]
> Please check if you have other deployments active during such time.

We didn't have any other ongoing deployments as far as I can tell from the logs.

**Comment by asmendez** [comment]
Hi, I have the same problem, since I have 2 applications in the same app service plan and I have both integrated with Azure devops through the Development Center. Even I have only one pipeline that deploys first to staging and after to production. I don't know what's wrong because a few days ago this was working well.

**Comment by sanchitmehta** [comment]
@asmendez Did the first deployment fail for some reason ? We have a cool off period in Deployment Locks (when an unexpected error occurs leading to a Kudu Crash) to avoid conflicts. Could you share your app name ?

 @mcdurdin  are you still facing this problem ?

**Comment by mcdurdin** [comment]
> @mcdurdin are you still facing this problem ?

We have not seen any errors in the last few deployments so I think it's good now.

**Comment by asmendez** [comment]
> @asmendez Did the first deployment fail for some reason ? We have a cool off period in Deployment Locks (when an unexpected error occurs leading to a Kudu Crash) to avoid conflicts. Could you share your app name ?
> 
> @mcdurdin are you still facing this problem ?

Hi, at the end I changed to a Windows App Service because in Linux this problem came up even with a fresh and new instance with the CD/CI. Even when I deleted all my instances from the same App Services Plan. I don't know why this was happening but I decided to change to a Windows PHP Web App and I don't have the problem now.

**Comment by evelo2** [comment]
Sorry, this issue is very much still alive.
Existing builds that were deploying successfully are now failing consistently.

I have posted the `YAML` to [StackOverflow](https://stackoverflow.com/questions/59741421/azure-yaml-deployment-failing-with-error-failed-to-deploy-web-package-to-app-s)

Any help would be appreciated.

**Comment by guiihlopes** [comment]
Constantly receiving this error.

**Comment by nmaarse** [comment]
I have the same problem :
Package deployment using ZIP Deploy initiated.
##[error]Failed to deploy web package to App Service.
##[error]To debug further please check Kudu stack trace URL : https://$dqipowebportal0001:***@dqipowebportal0001.scm.azurewebsites.net/api/vfs/LogFiles/kudu/trace
##[error]Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)

**Comment by guiihlopes** [comment]
I had the same problemas some days ago, and the issue is caused sometimes when the azure plan CPU reaches 100%. 

It gives conflict error message or this one:

```
An unknown error has occurred. Check the diagnostic log for details.
##[error]Failed to deploy web package to App Service.
```

**Comment by sanchitmehta** [comment]
@nmaarse @evelo2 This just means that Kudu was already processing a deployment request and to ensure that deployment is an atomic operation Kudu throws 409. We would have to see if the pipelines  was sending multiple requests. Could you open a support ticket (thereby you could share the app name) and we can investigate it further.

**Comment by nmaarse** [comment]
> 
> 
> @nmaarse @evelo2 This error implies that Kudu was already processing a deployment request and to ensure that deployment is an atomic operation Kudu throws 409. We would have to see if the pipelines was sending multiple requests. Could you open a support ticket (thereby you could share the app name) and we can investigate it further.

Hi sanchimehta. Happy to hear from you. Could you explain to me how and where to open a support ticket.

**Comment by wadeem** [comment]
same problem here

**Comment by charlatan154** [comment]
If you are on a free subscription plan, make sure you haven't reached your storage limit which is 1gb. If so, you have to upgrade to a paid one..

**Comment by i3anaan** [comment]
Had the same issue today. The service plan had plenty of room left (CPU/Memory/Disk), no other deployments were being processed at the time and a run that previously succeeded failed when retried today (same commit). Additionally it seemed that kudu crashed as well or something, the suggested link for more debug information gave a 500 and other attempts to access kudu, logs or files of the function all failed (the one I tried at least).

In the end I managed to resolve the issue by deleting the function app, and re-creating it (same settings). Thankfully this was a (new) function in a development environment, this would be an unacceptable workaround on a live production environment.

**Comment by JesseStolwijk** [comment]
Since today all our applications cannot be deployed, some applications receive a 500 while others receive a 409 response. Like @i3anaan pointed out deleting the function app and re-creating it resolves the issue. We cannot do this in prod since it will result in tens of minutes of down time

**Comment by jjavierdguezas** [comment]
same issue since today here, any solution other than recreating everything?

**Comment by jjavierdguezas** [comment]
I have solved the problem by scaling up the service plan where my azure functions were. I have switched from B2 to B3 tier, which only increase RAM and ACUs so I'm gessing that we had consumed all ACUs ? 
![image](https://user-images.githubusercontent.com/13673443/88804881-cfdb6c80-d1ae-11ea-91e8-f3d4ec39684c.png)

**Comment by IanYates83** [comment]
This also started happening for us today on S1 tier in multiple App Service Plans.
Scaling up to S2 tier solved it. After scaling back down to S1, I was still able to release.

Out typical ASP usage on S1 is 6% CPU and 60% memory.

**Comment by Kenneth-Abrams** [comment]
Experienced the same 409 error issue during deployment starting on 7/27 . Function apps on linux ASP.

**Comment by inatlewski** [comment]
I'm experiencing the same problem.
`2020-07-29T15:06:45.5681580Z ##[error]Failed to deploy web package to App Service.
2020-07-29T15:06:45.5701015Z ##[error]Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)`

**Comment by mahmoudhammouda** [comment]
hi, we faced to the same issue just now, maybe related to webdeply

**Comment by benoutram** [comment]
Experiencing the same problem here in a DevOps Release pipeline of App Service and Windows Function App projects. We first saw this error occur on 21/07/20 but seeing several occurrences today. The region for both the App Service and Functions  is West Europe.

```
2020-07-29T09:50:50.1288995Z Trying to update App Service Application settings. Data: {"WEBSITE_RUN_FROM_PACKAGE":"1"}
2020-07-29T09:50:50.4010044Z App Service Application settings are already present.
2020-07-29T09:51:46.6646202Z Package deployment using ZIP Deploy initiated.
2020-07-29T09:52:53.1693953Z ##[error]Failed to deploy web package to App Service.
2020-07-29T09:52:53.1707005Z ##[error]Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)
```

A possible workaround seems to be to change the deployment method to Zip deploy, but for one of our Application Services this has increased the deployment time from 1 minute to 21 minutes which is not going to be acceptable.

![image](https://user-images.githubusercontent.com/4147126/88821842-0a4b0680-d1bb-11ea-9a83-2c7378f9a968.png)

**Comment by jjavierdguezas** [comment]
Lets see if they can do something... https://twitter.com/jotajota930621/status/1288503705994764290

**Comment by pazcaldj** [comment]
Had this issue also several times today... Retried it a couple of minutes ago and succeeded... Seems like its fixed?

**Comment by mahmoudhammouda** [comment]
Hello , we faced this issue several time today and we used the  webdeploy
as workaround.

Le mer. 29 juil. 2020 à 19:19, Pascal de Jonge <notifications@github.com> a
écrit :

**Comment by zorge** [comment]
Having the same issue on S1, after upgrading the apps plan to a higher tier the issue is gone and the build succeeds. This is of course is not acceptable, Microsoft!

**Comment by suwatch** [comment]
I would be great if you all can share the webapp name - this would help us pinpoint.

**Comment by grantwar** [comment]
I'm having the same issue with multiple function apps, and they are hosted in a P2V2 and in another case in an S1.

**Comment by sanchitmehta** [comment]
@grantwar Are you still experiencing the same problem ? Could you provide us your app name indirectly ?

**Comment by grantwar** [comment]
> @grantwar Are you still experiencing the same problem ? Could you provide us your app name indirectly ?

IDFUAT-aldi-omegaconnector

**Comment by jjavierdguezas** [comment]
@suwatch I sent you an email with the information to the email in yout profile, but since I scaled up the app service we have not seen the error anymore

**Comment by sanchitmehta** [comment]
@jjavierdguezas Was your app deployed in West Europe ?

**Comment by jjavierdguezas** [comment]
> @jjavierdguezas Was your app deployed in West Europe ?

yes they are

**Comment by zorge** [comment]
Ive also had my service deployed in West Europe and after scale-up the error was gone. But I'd like to scale down and know what the error was caused by. Unfortunately I cannot give the app name here.

**Comment by suwatch** [comment]
Please follow this to report site name https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly

@jjavierdguezas  I did receive your message - but can't make anything out of the image.  Please share the site name (here or indirectly) as well as the UTC time of the incident.  We will investigate

**Comment by jjavierdguezas** [comment]
> Please follow this to report site name https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly
> 
> @jjavierdguezas I did receive your message - but can't make anything out of the image. Please share the site name (here or indirectly) as well as the UTC time of the incident. We will investigate

In my email I listed the six apps affected, 
<img src="https://user-images.githubusercontent.com/13673443/88860200-86653e80-d1fb-11ea-900f-3c62516189ad.png" height="300" />


The subscription id pattern: `58747***-****-****-****-***f432a26a1`

We noticed the incident on July 29 around 8:45 am UTC
We did the "fix" around 1:00 pm UTC

Let me know if you need more info

**Comment by suwatch** [comment]
From our telemetry, we observed below exception happening to multiple sites and vms started a few days ago.   We are unable to repro.   If anyone is still stuck in this state, please try to go to portal, stop and, wait 10s, start the site  (assuming you can afford site to be down for such short period).   Try zipdeploy again - and see if that mitigate the issue.

```csharp
Error occurred type="error" text="The type initializer for '<Module>' threw an exception." stackTrace="   at Newtonsoft.Json.Converters.BinaryConverter.CanConvert(Type objectType)
   at Newtonsoft.Json.JsonSerializer.GetMatchingConverter(IList`1 converters, Type objectType)
   at Newtonsoft.Json.Serialization.DefaultContractResolver.InitializeContract(JsonContract contract)
   at Newtonsoft.Json.Serialization.DefaultContractResolver.CreateObjectContract(Type objectType)
   at Newtonsoft.Json.Serialization.DefaultContractResolver.CreateContract(Type objectType)
   at Newtonsoft.Json.Serialization.DefaultContractResolver.ResolveContract(Type type)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
   at Newtonsoft.Json.Linq.JToken.FromObjectInternal(Object o, JsonSerializer jsonSerializer)
   at Newtonsoft.Json.Linq.JObject.FromObject(Object o, JsonSerializer jsonSerializer)
   at Kudu.Core.Infrastructure.LockFile.WriteLockInfo(String operationName, Stream lockStream) in C:\Kudu Files\Private\src\master\Kudu.Core\Infrastructure\LockFile.cs:line 213
   at Kudu.Core.Infrastructure.LockFile.Lock(String operationName) in C:\Kudu Files\Private\src\master\Kudu.Core\Infrastructure\LockFile.cs:line 153" innerText="Attempted to read or write protected memory. This is often an indication that other memory is corrupt." innerStackTrace="   at <CrtImplementationDetails>.LanguageSupport.Initialize(LanguageSupport* )
   at .cctor()
```

**Comment by suwatch** [comment]
### Update

We are currently investigating.   It is likely due to recent update on the VMs leading to this.   It affects "**West Europe**", "**West US 2**", "**North Central US**" and "**West Central US**" regions.    We are current working on mitigation across in the next few hours.   

You can also mitigate yourself by performing either of the below.

1) Restarting Kudu process.   Browse to SCM site -> Process Explorer and Kill w3wp (SCM).    The new w3wp will start up automatically.   If you have multiple instances, you may need to use instance dropdown to iterate and do the same on all instances.

2) Goto portal.  Navigate to site blade, Stop, wait 5s and Start.   This unfortunately will cause short down time for the site.

3) Scale up and scale down.   This indirectly restarts SCM sites.

We are also gradually mitigate in the next few hours.  Apology for the inconvenience.

**Comment by i3anaan** [comment]
@suwatch Good to know there is being worked on an a fix!
Would it still help you if we shared more information about the apps being affected?

I think this information should give you the ability to pinpoint one of our function apps affected by this issue:
- 2020-07-30T08:02:12Z   [Information] Id=eec0c97e-8138-4bd9-81fe-cd233ec4dedf)

(The guide/steps in the [docs](https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately) seems lightly outdated)

**Comment by jjavierdguezas** [comment]
Today we faced this issue in another subscription, this time I restarted the app (as @suwatch said) and then the release went just fine. So, no scaling is really needed... good news to save money... 😅

**Comment by julen-folky** [comment]
I had the issue too.
I could not kill the w3wp process because Kudu did not show anything in the process' list.
Restarting the site did not work.
I stop the site, deploy again (using Azure DevOps pipelines), start again. Worked!
Then I tried a new deploy without stopping the site. Also works.

The problem and steps above executed on staging slot. Production slot was not affected by the issue.

**Comment by lewis-harris** [comment]
I had the issue too, in west europe..
I could not kill the w3wp process because Kudu did not show anything in the process' list.
Restarted the site and deployed again succesfully

**Comment by julen-folky** [comment]
FWIW, @lewis-harris I have the site in west europe too.

**Comment by remcorakers** [comment]
We received the same issue since Friday last week for one of our App Services in West Europe. We tried to do the following:

1. Restart the App Service and re-deploy via our Azure build pipeline => still received the 409 error
2. Stop the App Service re-deploy via our Azure build pipeline => still received the 409 error
3. Kill all w3wp processes in the SCM console and re-deploy via our Azure build pipeline => still received the 409 error
4. Delete the App Service, re-deploy the App Service via our ARM templates, stop the App Service, and re-deploy via our Azure build pipeline => success!

So only deleting the entire App Service and re-deploying fixed the issue.

**Comment by kayhantolga** [comment]
Had the same issue
** Scaled up app service, stop and wait a bit, run again, run the deployment => worked fine

I tried scaling up and down then deployment, didn't work for me.

**Comment by yoelOhayon** [comment]
Same issue , 
could not fill the w3wp process because Kudu did not show anything in the process list.
restart the app changed nothing

I added a new slot and configure my Azure Devops Release to the new slot and it worked !

**Comment by sympthom** [comment]
This is all of the sudden a problem for us (this has not been a problem since 1.5 years). Why do we get "Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)" when deploying to WE using zip deploy? How can we do to fix it? This is not a descibable error message.

**Comment by suwatch** [comment]
@sympthom   did you try the mitigation mentioned above?

**Comment by atmike** [comment]
I have the same problem.
but i could solve it by stopping the app-service, starting it again and then restarting it 3 times  in quick succession. This seems to stop the w3wp and the release is working again.

**Comment by Gismork** [comment]
We started experiencing this issue on one App Service on July, 29th hosted in West Europe.
We are now seeing this issue on other App Service.
 
We were able to workaround this issue by killing the Kudu process and redeploying our application again on the App Service. We use Azure DevOps to deploy our applications (using the Run from package deployment mode).

**Comment by abercea** [comment]
> I have the same problem.
> but i could solve it by stopping the app-service, starting it again and then restarting it 3 times in quick succession. This seems to stop the w3wp and the release is working again.

This also worked for me (azure func on win, run from package deploy)

**Comment by paulhickman-a365** [comment]
I have a pipeline that sequentially deploys:
 - 4 .net core azure function apps to app service plan 1 on elastic premium
 - 4 .net core web applications to app service plan 2 on s2
 - 4 static html sites to app service plan 3 also on s3

All are using zip deploy, and we intermittently get this error on any of the 12 apps. We can usually work around it by stopping the failing app and then re-running the deployment.

**Comment by Tasboo** [comment]
I ran into the same error when deploying a function app. "Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)"

After several failed attempts, we restarted the function app through the Azure portal. After it the function app finished restarting the deploy finally worked.

**Comment by timg83** [comment]
I also ran into the Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)" today. Kudu did not respond when clicking Debug Console > CMD. Got the "Azure Functions Runtime is unreachable" notification on the Azure Function overview.

What did work for us was to create a new slot, deploy to that slot, swap it to production. After this I removed the new slot and redeployed without using slots and this was successful.

**Comment by pietersap** [comment]
Confirmed that scaling the app service plan up and down works. Could not kill the w3wp process since the process explorer in Kudu wouldn't load.

Scaled up my B1 app service plan to B2, did the deployment and succeeded, scaled down to B1 and afterwards the deployments continued succeeding.

**Comment by pietersap** [comment]
edit: killing w3wp processes also works

**Comment by arynaq** [comment]
Any updates here? Is less than ideal uncertainty for deployment of business critical software.

**Comment by suwatch** [comment]
Are you repeatedly running into issue?  If so, do share the [App Service name](https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly).    The expected is you should run into it once.   As for rollout, we are currently pushing out the fix.  It should be done by end of the month.

**Comment by arynaq** [comment]
> Are you repeatedly running into issue? If so, do share the [App Service name](https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly). The expected is you should run into it once. As for rollout, we are currently pushing out the fix. It should be done by end of the month.

We managed to circumvent this by doing the scale up/down approach. It is not a persistent issue but we have many different apps where this has happened and we don't always have direct access to them (outsourced ops) to do this.

It is just an inconvenience/annoyance  that costs time (therefore money) though the workarounds work.

**Comment by suwatch** [comment]
Totally understand the pain and appreciate the patience.   We will continue to roll out fix to address this - ETA end of Aug 2020.

**Comment by erwelch** [comment]
I'm seeing  this too, but only on our applications with these settings:

```
WEBSITE_DNS_SERVER with value 168.63.129.16
WEBSITE_VNET_ROUTE_ALL with value 1
```

Is this related or a separate issue? Seems if I remove those settings then deploy works fine (though our app doesn't).

**Comment by J0rgeSerran0** [comment]
Same issue here

Linux machine running an ASP.NET Core 2.2 web app at West Europe

The Release Azure DevOps process to deploy the App on Azure was running successfully, but suddenly we have received an error during the automatic deploy process that is being executed on Azure DevOps: 

_##[error]Failed to deploy web package to App Service.
##[error]Error: Error: Failed to deploy web package to App Service. Conflict (CODE: 409)
##[warning]Error: Failed to update deployment history. Error: Bad Request (CODE: 400)_

**My steps to fix the problem:**
- Scale up the App Service Plan on Azure
- Execute from Azure DevOps the Release process manually
- Scale out the App Service Plan again.

This thread and comments are correct and saved my time and problems in a critical issue.

Thanks all.

**Comment by NickTolhurst26** [comment]
We're also experiencing this issue and are currently manually invoking the scale-up/scale-down workaround for a couple of our client's production apps.

@suwatch - Was the fix for this rolled out on time?

**Comment by tomkuijsten** [comment]
2 weeks further, we got plenty of 409's today, is the fix on all envs @suwatch ?

**Comment by suwatch** [comment]
Umm .. you might be running into different issue.  could you share a few incidents, [site names](https://github.com/projectkudu/kudu/wiki/Reporting-your-site-name-without-posting-it-publicly) and UTC time of operations?

**Comment by panuoksala** [comment]
I'm having the same problem and scaling from S1 to S2 solved the issue... This is one easy way to make more money.

**Comment by suwatch** [comment]
It would be best to create a new gissue with details for us to help investigate.   As far as the main regression goes, it only impacted Windows and fix were rolled out completely weeks ago.   Yours must be something new or transient issue.

**Comment by delroyh** [comment]
I'm having this issue all of a sudden. Deploys were fine for forever (at least the last year), then all of a sudden starting to see this. A quick google search brought me to this thread.
Repeated deployment attempts _eventually_ succeed. I haven't tried the scale up and down solution as a work around yet.
If I find some motivation in the next few days I'll see if I can gather some more details to throw in the pile here.

**Comment by suwatch** [comment]
I would suggest to open a new gissue with all the necessary info.

**Comment by ikuosaito1989** [comment]
This problem occurred on East Japan and B1 instance of Azure App Service on Linux. I would like to know how to solve this problem other than scaling up.

More information can be found in the following issue.

https://github.com/Azure/webapps-deploy/issues/98

**Comment by JV-conseil** [comment]
## Error: Failed to update deployment history. Conflict (CODE: 409)

- Location: **West Europe**
- App Service Plan: **P2v2: 1**
- Operating System: **Linux**
- Build: **GitHub Action**

```bash
[group]Run azure/webapps-deploy@v2
with:
  app-name: some-app-name
  slot-name: dev
  publish-profile: ***
  package: .
env:
  pythonLocation: /opt/hostedtoolcache/Python/3.8.7/x64
[endgroup]
Package deployment using ZIP Deploy initiated.
[error]Failed to deploy web package to App Service.
[error]Deployment Failed with Error: Error: Failed to deploy web package to App Service.
Internal Server Error (CODE: 500)
[warning]Error: Failed to update deployment history.
Conflict (CODE: 409)
```

**Comment by nertim** [comment]
@JV-conseil,

Are you still running in to this issue? If so, can you please open a new GitHub issue with following information:
1. Site/slot names
2. latest Location, OS
3. latest GitHub Action log

A new GitHub issue will help us manage investigation of the specific issues a little better.

**Comment by delroyh** [comment]
> I'm having this issue all of a sudden. Deploys were fine for forever (at least the last year), then all of a sudden starting to see this. A quick google search brought me to this thread.
> Repeated deployment attempts _eventually_ succeed. I haven't tried the scale up and down solution as a work around yet.
> If I find some motivation in the next few days I'll see if I can gather some more details to throw in the pile here.

I wanted to circle back on this (after getting some notifications about these recent comments on the thread)
My issues were resolved by fixing a completely unrelated issue that _happened_ to manifest this way. It's been...a while...since I got it fixed and can't for the life of me remember the specifics, other than it had absolutely nothing to with Kudu despite the issue manifesting as a Kudu deploy failure.
So, not the most useful update, but I don't have problems anymore 😂

**Comment by vip32** [comment]
ran into the issue today, gosh (linux webap, B2)!? scaling up to B3 really did the trick

**Comment by martinlarosa** [comment]
I believe that's pretty random @vip32 
I sometimes fixed it like that but then also made it work with lower tiers. Also had it fail with higher tiers.

I "fixed"it completely by going to the pipeline and forcing it to use web deployment:
![image](https://user-images.githubusercontent.com/7032277/107217723-576f2700-69ed-11eb-93c7-ffb9d1b3c914.png)

**Comment by emmanuelvr** [comment]
I had this issue and all I did was go into the Deployment Center, click on "Disconnect" and then re-deploy

**Comment by nikidandwani** [comment]
Ran into the same issue today in west europe!

**Comment by adriannasui** [comment]
Still happenning to us on linux app service

**Comment by adriannasui** [comment]
ok. Found the issue - it is related to our app service plan storage limit - we were right at the limit of 10GB due to some logs. 

> Still happenning to us on linux app service

**Comment by scottrudy** [comment]
We just ran into this issue on a production service and managed to resolve it in the short term. There are few things going on. First, there is a [storage limit](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#app-service-limits) across all app services in a region and resource group. In our case we hit 500 GiB across all the app service plans. Our plans were all P3v2, so we had no ability to scale up/down when this happened, as we were at the highest tier.

In addition, we were trying to run multiple DevOps pipelines to multiple applications at the same time. When a deployment happens some storage is allocated to allow for the deployment to complete. In our case this allocation was enough to put us over the limit. We found if we continually monitored the storage quota across the region and resource group it eventually freed up enough storage where we were able to deploy.

Our long term fix is to identify the applications that are using a large amount of storage and determine how we can either limit the amount of storage each one of them is using, or provision some of those applications into new resource groups so the limit isn't a concern.

**Comment by YitongFeng-git** [comment]
> @asmendez Did the first deployment fail for some reason ? We have a cool off period in Deployment Locks (when an unexpected error occurs leading to a Kudu Crash) to avoid conflicts. Could you share your app name ?
> 
> @mcdurdin are you still facing this problem ?

We are facing the same issue.
If the first time the deploy failed due to 502, and we retry immediately, it will keeps failed in 409.
If it has cool off period, what's the expect time of this time?

**Comment by viniciushss-interstar** [comment]
I'm having this issue right now. Linux App Service @ East US.

**Comment by gasserma** [comment]
I'm also having this issue right now. East US Windows app service.

**Comment by nephinj** [comment]
Having this issue frequently.  Seems random when it works or not.  Tried increasing to P3V2 sku and still no luck.

**Comment by matttrakker** [comment]
I'm having it since yesterday on Linux based functions in west europe, no consumption, B2/ 1-3 instances makes no difference

**Comment by aderici** [comment]
For everyone having this problem, please set SCM_DO_BUILD_DURING_DEPLOYMENT to false in your app service settings and thank me later

**Comment by worksofliam** [comment]
I am having this problem right now.

**Comment by matttrakker** [comment]
My issue has been solved. The problem has been very fast chained deployments.
Then one has still been running and has blocked the newer deployment.

You can also find informations in the logs regarding the running and the blocked deployment.

Initial releasing the block has been done by upgrading the plan, adding a slot, wait until it has warmed up, then swap it. Now redeploy your code, delete the old slot and down scale the plan again.

hope that helps anybody 😊

I still have the opinion that in a devops world, the system should have a wait process implemented for such cases.

**Comment by marlenabowen** [comment]
My deployments are passing, but I am continuously seeing the `Error: Failed to update deployment history. Error: Conflict (CODE: 409)` warning. In the line before that in my logs, I see `{"Message":"There is an auto swap deployment currently ongoing please try again when it completes."` I can't figure out what deployment is ongoing. I am using auto swap and so my pipeline does not run any commands specific to the deployment slot, besides actually deploying to it. I have tried playing around with adding the "Complete Swap" task to my deploy job, but it doesn't seem to make a difference. Anyone else encounter this issue with a successful deployment but a warning about failing to update deployment history?

**Comment by k290** [comment]
This happened for us for Linux function apps, when the CI/CD is released in quick succession and possibly overlapping in parallel. Although this issue doesn't happen with our windows function apps, our workaround was to make sure the CI/CD batches changes to avoid this quick succession of builds and releases. Below is the yaml
```
trigger:
  batch: true
  branches:
    include:
      - develop
 ```

**Comment by oscarhermoso** [comment]
Ensure that your default and staging slots have different values for `WEBSITE_CONTENTSHARE` in app configuration

https://github.com/microsoft/azure-pipelines-tasks/issues/12354#issuecomment-655986449

**Comment by petro2050** [comment]
I have this issue every single day. It's become quite annoying. I agree that deployment logs are useless.

**Comment by matttrakker** [comment]
@petro2050 we found two possible solutions, which have solved it since a while:
• do not deploy very fast
• ensure that if you run on a service plan the config of the asp is good enough as the deployment seems to need more than just running the code

perhaps that helps 😊

**Comment by ransagy** [comment]
For us, We switched to the "run from package" method, Which seemed to make these all but disappear. More of a workaround, but hey..

**Comment by mcdurdin** [comment]
This has cropped up for us again. It seems that large deployments are just not really reliable with Azure Web Apps. Deployment logs don't really help either. We have thousands of files in the deployment, but usually only one or two files have changed when we deploy, and it looks like Kudu just doesn't cope well with that scenario.

**Comment by mcdurdin** [comment]
We tried changing to a GitHub action with zip deployment. That didn't fix the problem, which is that the kudu part of the deployment is very slow.

So we have moved over to our own deployment method which bypasses Kudu altogether. It uses a GitHub action that pings a webhook on our site, which triggers a web job that runs `git pull` from a GitHub repo. This means having the .git folder in the website root (which we then block access to with .htaccess/web.config rules). This is maybe a bit of string and rubber bands, but this has reduced the deployment time from >1hr down to under 1 minute, for a site with around 50,000 files on it. So far, we've seen no errors.

Another advantage is if we do get an error, it's easy to `git pull` or `git checkout .` or even `git clean -fdx`(!) to cleanup from a console on the web service and return our site to a precise match of the source repo.

See https://github.com/keymanapp/s.keyman.com/pull/55 for our implementation.

Given we are no longer experiencing this problem, I guess I should close this issue! Hoping this information helps others who are currently blocked with this.

**Comment by EsgeHansen** [comment]
Had this issue today as well in west europe.
It deployed fine on our test subscription, but as i tried to deploy it in our production subscription it failed saying that the requested features were not supported. After reading ll of this i tried changing the plan from S1 to S2 and then it deployed right away, and could update the plan back to S1 after.

**Comment by liamfoneill** [comment]
It's worth noting for us the conflict was due to the service hook in Azure DevOps sticking around from when deployment was setup previously. We needed to manually delete it in Azure DevOps in order to get the conflict error to go away

**Comment by SellamiHabib** [comment]
I stopped the correspondent Azure web service and forgot about it, I solved the issue by starting it.

**Comment by teslavitas** [comment]
Same issue in west Europe, deployment is either successful after 30+ minutes or crashed with  error 409. Several days ago everything worked in less then a minute. I use a free tier, app service name is myhouse-jobs-test

**Comment by matthiasschuurmans** [comment]
Experiencing this issue as well. Bit of context: Python FA, so we have to use a Linux plan. 

- Initially, we had a Linux consumption plan and ran from package with a URL (only way that is supported for Linux Consumption according to [this](https://learn.microsoft.com/en-us/azure/azure-functions/run-functions-from-deployment-package#enable-functions-to-run-from-a-package)).
- Due to reasons, we switched to a dedicated plan, S1
- We started getting this error, and diving in the logs, I came across `Run-From-Zip is set to a remote URL using WEBSITE_RUN_FROM_PACKAGE or WEBSITE_USE_ZIP app setting. Deployment is not supported in this configuration.` Seems weird, it doesn't say that Linux with dedicated plan does not support URL, but ohwell
- We switched to `WEBSITE_RUN_FROM_PACKAGE=1`, but that causes the wwwroot folder to become readonly, which doesn't work for us
- So in the end we switched to just `zipDeploy` in our [AzureFunctionApp@2](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/azure-function-app-v2?view=azure-pipelines) CI/CD pipeline. It started taking quite a long time (~15 minutes, big zip, I think because the zip contains the complete python environment including all dependencies, so around 18k files), but in the end works
- ... Sometimes. And sometimes we still get this error

Microsoft warns about zipDeploy being a bit less atomic than runFromPackage ([1](https://github.com/Azure/app-service-announcements/issues/84) [2](https://github.com/Azure/app-service-announcements/issues/84)), so maybe that's also where the 409 conflict comes from? The deploy might be OK and finished, but maybe under the hood there is still something happening that causes a fast next deployment to fail?
